src/common/meta/src/key/table_region.rs

```diff
@@ -68,6 +71,67 @@ impl TableRegionManager {
             .transpose()
     }
 
+    pub async fn get_old(&self, table_name: &TableName) -> Result<Option<TableRegionValue>> {
+        let key = TableGlobalKey {
+            catalog_name: table_name.catalog_name.clone(),
+            schema_name: table_name.schema_name.clone(),
+            table_name: table_name.table_name.clone(),
+        }
+        .to_string();
+        let raw_key = key.as_bytes();
+
+        self.kv_backend
+            .get(raw_key)
+            .await?
+            .map(|kv| TableGlobalValue::from_bytes(kv.value()))
+            .transpose()
+            .map(|v| {
+                v.map(|v| TableRegionValue {
+                    region_distribution: v.regions_id_map.into_iter().collect(),
+                    version: 0,
+                })
+            })
+            .context(InvalidCatalogValueSnafu)
+    }
+
+    pub async fn put_old(
+        &self,
+        table_name: &TableName,
+        region_distribution: RegionDistribution,
+    ) -> Result<()> {
+        let key = TableGlobalKey {
+            catalog_name: table_name.catalog_name.clone(),
+            schema_name: table_name.schema_name.clone(),
+            table_name: table_name.table_name.clone(),
+        }
+        .to_string();
+        let raw_key = key.as_bytes();
+
+        let table_info = self
+            .kv_backend
+            .get(raw_key)
+            .await?
+            .map(|kv| TableGlobalValue::from_bytes(kv.value()))
+            .transpose()
+            .context(InvalidCatalogValueSnafu)?
+            .map(|v| v.table_info)
+            .with_context(|| InvalidTableMetadataSnafu {
+                err_msg: format!("table global value for {table_name} is empty"),
+            })?;
+
+        let raw_value = TableGlobalValue {
+            node_id: 0,
+            regions_id_map: region_distribution.into_iter().collect(),
+            table_info,
+        }
+        .as_bytes()
+        .context(InvalidCatalogValueSnafu)?;
+
+        let req = PutRequest::new().with_key(raw_key).with_value(raw_value);
+        self.kv_backend.put(req).await?;
```

Comment:
Is there any issue with concurrent updates when using get and put, considering that they do not have atomic execution?

Reply:
The intention was not make a huge PR. Concurrency concerns are address in the following changes.